### PR TITLE
Update function docs

### PR DESCRIPTION
**Pull request by TrimarcJake**
Used Copilot to add comment-based help to everything. It's not the best, but it's better than nothing!

### CODE REVIEW

**Comment by TrimarcJake** [comment]
Issue #115 supports this.

File: Private/ConvertFrom-IdentityReference.ps1
Comment (SamErde) [inline Private/ConvertFrom-IdentityReference.ps1:1]: CBH looks good, with one nitpick: the indentation isn't consistent with other files' indentation.

File: Private/Export-RevertScript.ps1
Comment (SamErde) [inline Private/Export-RevertScript.ps1:1]: The synopsis and description should say, "This _creates_ a script that reverts changes performed by Locksmith."

File: Private/Find-ESC8.ps1
Comment (SamErde) [inline Private/Find-ESC8.ps1:1]: Do you want to align the ESC8 synopsis with the prior ones ("This script finds AD CS (Active Directory Certificate Services) objects that have the ESC8 vulnerability.") or is this one a unique case given the different types of checks that could be performed?

File: Private/New-OutputPath.ps1
Comment (SamErde) [inline Private/New-OutputPath.ps1:1]: You have a bad copy/paste in this one: the CmdletBinding and param() blocks are in there twice. Indentation for CBH also isn't consistent with the standard that [most] other functions have followed.

File: Private/New-OutputPath.ps1 [hunk 1]
```diff
@@ -1,4 +1,31 @@
...
+    .PARAMETER OutputPath
+    Specifies the base path where the output directories will be created.
+
+    .EXAMPLE
+    New-OutputPath -Targets "Forest1", "Forest2" -OutputPath "C:\Output"
+    This example creates two output directories named "Forest1" and "Forest2" under the "C:\Output" directory.
+
+    #>
+
+    [CmdletBinding(SupportsShouldProcess)]
```
Comment (SamErde) [inline Private/New-OutputPath.ps1:22]: Remove extra CmdletBinding and param() block.

File: Private/Invoke-Scans.ps1 [hunk 2]
```diff
@@ -1,4 +1,62 @@
...
+
+    .NOTES
+        - The script requires the following functions to be defined: Find-AuditingIssue, Find-ESC1, Find-ESC2, Find-ESC3Condition1,
+          Find-ESC3Condition2, Find-ESC4, Find-ESC5, Find-ESC6, Find-ESC8.
+        - The script uses Out-GridView or Out-ConsoleGridView for interactive selection when the 'PromptMe' scan option is chosen.
+        - The script returns a hash table containing the results of the scans.
+
+    .EXAMPLE
+        # Perform all scans
+        Invoke-Scans.ps1
```
Comment (SamErde) [inline Private/Invoke-Scans.ps1:31]: Remove '.ps1' from the example.

File: Private/Invoke-Scans.ps1 [hunk 3]
```diff
@@ -1,4 +1,62 @@
...
+    .EXAMPLE
+        # Perform only the 'Auditing' and 'ESC1' scans
+        Invoke-Scans -Scans 'Auditing', 'ESC1'
+
+    .EXAMPLE
+        # Prompt the user to select the scans to perform
+        Invoke-Scans -Scans 'PromptMe'
+    #>
+
+    [CmdletBinding()]
```
Comment (SamErde) [inline Private/Invoke-Scans.ps1:42]: Added an extra CmdletBinding somehow. (Go home you're drunk.)

File: Private/Invoke-Scans.ps1 [hunk 4]
```diff
@@ -1,4 +1,62 @@
...
+    param (
```
Comment (SamErde) [inline Private/Invoke-Scans.ps1:43]: Added an extra param () block. (Friends don't let friends...)

File: Private/Set-AdditionalCAProperty.ps1 [hunk 5]
```diff
@@ -1,4 +1,42 @@
...
+    .EXAMPLE
+        $ADCSObjects = Get-ADObject -Filter { objectClass -eq 'pKIEnrollmentService' }
+        Set-AdditionalCAProperty -ADCSObjects $ADCSObjects
+
+    .NOTES
+        Author: Jake Hildreth
+        Date: July 15, 2022
+    #>
+
+    [CmdletBinding(SupportsShouldProcess)]
```
Comment (SamErde) [inline Private/Set-AdditionalCAProperty.ps1:28]: Re-paste? Extra CmdletBinding and param again.

**Review by SamErde — CHANGES_REQUESTED** [review]
Several of the functions had redundant `CmdletBinding` and `param ()` blocks added in this PR and need to be removed carefully.

Style question: what is a good way that we can distinguish commands, their output, and the description text for example blocks? I do like the way that most seem to have a blank link between the example command and the example description. Others have no distinguishing paragraph breaks between the example, the example output, and the description.

**Comment by TrimarcJake** [comment]
> Several of the functions had redundant `CmdletBinding` and `param ()` blocks added in this PR and need to be removed carefully.

Removed! If they are all gone to your satisfaction, please to be approving.

> Style question: what is a good way that we can distinguish commands, their output, and the description text for example blocks? I do like the way that most seem to have a blank link between the example command and the example description. Others have no distinguishing paragraph breaks between the example, the example output, and the description.

We can talk about this in Friday's meeting and decide on a standard. And if there's a definite style guide we want to follow, pls lmk!

**Review by SamErde — APPROVED** [review]
LGTM!

**Pull request merged**